src/components/Header/components/QuickSearch/index.tsx

```diff
@@ -80,9 +80,9 @@ const performSearch = debounce(
 
 type Props =
   | {
-      disableLink: true;
-      customPlaceholderText: string;
-      onSearchSubmit: (person: SearchResult) => void;
+      disableLink?: true;
+      customPlaceholderText?: string;
+      onSearchSubmit?: (person: SearchResult) => void;
```

Comment:
Why did these get marked as optional? `customPlaceholderText` and `onSearchSubmit` are required when `disableLink` is `true`.

Reply:
![image](https://github.com/gordon-cs/gordon-360-ui/assets/100231490/255b10fd-816b-4db6-ab16-0eb6874c018c)
When I mark them as required this line of code complains, is there a way to resolve this nicely?

Reply:
The solution is to mark them as optional in the `disableLink: false` case of the union (and also mark `disableLink` itself as optional). This tells TypeScript that the props are not required to be set in the case that `disableLink` is either `false` or `undefined`. I guess typescript doesn't allow a prop that can only be `undefined` to not be passed, you have to explicitly mark that prop as optional as well.

The other way to handle this would be to remove the two undefined props from the case where `disableLink` is false or undefined. However, if we do that, we can't destructure those props, since they only exist when `disableLink` is true. We could handle that by spreading the props other than `searchFunction` into a `props` object:
```typescript
const QuickSearch = ({searchFunction, ...props}: Props) => {...}
```
And then we'd have to check `props.disableLink` before referencing the other two props. e.g. 
```javascript
const linkProps = props.disableLink
      ? { onClick: () => props.onSearchSubmit!(person) }
      : {
          component: Link,
          to: `/profile/${person.UserName}`,
        };
```

Personally, I think it's easier to just mark the other two props as optional when `disableLink` isn't `true`.